gammapy/estimators/map/ts.py

```diff
@@ -468,6 +469,8 @@ def run(self, dataset):
                 * flux_ul : upper limit map
 
         """
+        if isinstance(dataset, MapDatasetOnOff):
```

Comment:
As the issue is the necessity to use the `cash` statistic, you could test `dataset.stat_type != "cash"`

Reply:
Done !
Should I add "with stat type {dataset.stat_type}" at the end of the error print or is it good enough as it is?